syntax = "proto3";

import "google/protobuf/empty.proto";
package wemmick;

service Wemmick {
  rpc HelloWorld (google.protobuf.Empty) returns (Response) {};
  rpc ListDataSources (google.protobuf.Empty) returns (DataSourceList) {};
  rpc ListExpectationSuites (google.protobuf.Empty) returns (ExpectationSuitesList) {};
  rpc CreateExpectationSuiteFromJsonSchema (CreateExpectationSuiteRequest) returns (Response) {};
  rpc CreateExpectationSuiteFromAvroSchema (CreateExpectationSuiteRequest) returns (Response) {};
}

message Request {}

message Response {
  string message = 1;
}

message DataSource {
  message Credentials {
    string connection_string = 1;
  };

  string name = 1;
  string class_name = 2;
  string module_name = 3;
  Credentials credentials = 4;
}

message DataSourceList {
  repeated DataSource data_sources = 1;
}

message ExpectationSuitesList {
  repeated string expectation_suites = 1;
}

message CreateExpectationSuiteRequest {
  string file_path = 1;
  string suite_name = 2;
}